libraries/rig-package/README.md

```diff
@@ -0,0 +1,199 @@
+
+  "rigPackageName": "example-rig",
+  "rigProfile": "sdk-library"
+}
+```
+
+This eliminates the `"extends"` stub files entirely.  A tool that implements the `rig.json` system would
+look for its config file (`config/<targetFile>.json`) using this procedure:
+
+1. First check for `config/<targetFile>.json` in the project folder; if found, use that file.  OTHERWISE...
+2. Check for `config/rig.json`; if found, then this project is using a rig package.  Read the `<rigPackageName>`
+   and `<rigProfile>` settings from that file.
+3. Use Node.js module resolution to find the `<rigPackageName>` package folder (let's call that `<rigPackageFolder>`)
+4. Check for `<rigPackageFolder>/profile/<rigProfile>/<targetFile>.json`; if found, use that file.  OTHERWISE...
+5. If the `<targetFile>.json` cannot be found in either of these places, the behavior is left to the tool.
+   For example, it could report an error, or proceed using defaults.
+
+This eliminates the need for most config files that support this system.  In cases where we need a project-specific
+customization, the `"extends"` field is still supported.  For example, **project1** can still add a custom setting
+like this:
+
+**project1/config/api-extractor.json**
+```js
+{
+  "$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
+  "extends": "example-rig/profile/library/api-extractor.json",
+
+  // Custom setting:
+  "mainEntryPointFilePath": "<projectFolder>/lib/custom.d.ts",
+}
+```
+
+The result is a much shorter inventory of files:
+
+```
+example-rig/package.json
+example-rig/profile/library/api-extractor.json
+example-rig/profile/sdk-library/api-extractor.json
+
+project1/package.json
+project1/config/rig.json
+project1/config/api-extractor.json  <-- custom setting shown above
+project1/src/index.ts
+
+project2/package.json
+project2/config/rig.json
+project2/src/index.ts
+
+project3/package.json
+project3/config/rig.json
+project3/src/index.ts
+```
+
+
+## A helper library
+
+The `@ruhstack/rig-package` library provides a lightweight API for loading the `rig.json` file and performing lookups.
+The package does not depend on any other NPM packages.  It also includes the JSON schema file `rig.schema.json` for
+tools that need it.
+
+Example usage of the API:
+
+```js
+import * as resolve from 'resolve';
+import { RigConfig } from '@rushstack/rig-package';
+
+// Probe for the rig.json file and load it if found
+const rigConfig: RigConfig = RigConfig.loadForProjectFolder({
+  // Specify a  project folder (i.e. where its package.json file is located)
+  packageJsonFolderPath: '/path/to/project3',
+
+  // If you want to use the RigConfig.getResolvedProfileFolder() API, you need to provide
+  // a Node.js module resolver.  For portability, @rushstack/rig-package does not depend on one.
+  moduleResolver: (options: IModuleResolverOptions): string => {
+    return resolve.sync(options.modulePath, { basedir: options.baseFolderPath });
+  }
```

Comment:
Is this compatible with the `node-core-library` `Import.resolveModule` API? It'd be great if it was.

Reply:
It sure is